packages/client/scripts/default-index.js

```diff
@@ -25,10 +25,12 @@ module.exports = {
     defineExtension,
     getExtensionContext,
     get prismaVersion() {
-      // default-index might gets copied to a .prisma/client directory and
-      // no longer be able to require @prisma/engines-version package. Delegating
-      // version information to the file at stable location allows us to overcome this.
-      return require('@prisma/client/version')
+      const { version, dependencies } = require('@prisma/client/package.json')
+
+      return {
+        client: version,
+        engine: dependencies['@prisma/engines-version'].split('.')[3],
```

Comment:
I don't think that matches generated version for integration branches. Version in the `package.json` contains branch name, while `prismaVersion.engine` is always just a commit hash.

Reply:
I tried to move everything together because it makes things simpler for latter ESM support. Now applied a solution to your suggestion.